cheddar/cheddar-events/src/main/java/com/clicktravel/cheddar/event/EventMessageHandler.java

```diff
@@ -40,20 +42,26 @@ public void handle(final TypedMessage typedMessage) throws MessageHandlingExcept
         try {
             final String messageType = typedMessage.getType();
             logger.debug("Handling: " + messageType);
-            final EventHandler<E> eventHandler = eventHandlers.get(messageType);
-            final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
-            if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
-                throw new IllegalStateException("Event of type " + event.getClass() + " is not compatible with "
-                        + eventHandler.getEventClass() + " in event handler");
+            for (final EventHandler<E> eventHandler : eventHandlers.get(messageType)) {
+                final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
+                if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
+                    throw new IllegalStateException("Event of type " + event.getClass() + " is not compatible with "
+                            + eventHandler.getEventClass() + " in event handler");
+                }
+                eventHandler.handle(event);
```

Comment:
@clicktravel-robin  You've identified a bug here, an exception throwing handler should not prevent other handlers running. I'll update this pull request.